src/mds/PurgeQueue.cc

```diff
@@ -28,9 +28,22 @@ static ostream& _prefix(std::ostream *_dout, mds_rank_t rank) {
   return *_dout << "mds." << rank << ".purge_queue ";
 }
 
+const std::map<std::string, PurgeItem::Action> PurgeItem::actions = {
+  {"NONE", PurgeItem::NONE},
+  {"PURGE_FILE", PurgeItem::PURGE_FILE},
+  {"TRUNCATE_FILE", PurgeItem::TRUNCATE_FILE},
+  {"PURGE_DIR", PurgeItem::PURGE_DIR}
+};
+
 void PurgeItem::encode(bufferlist &bl) const
 {
-  ENCODE_START(1, 1, bl);
+  ENCODE_START(2, 2, bl);
+  encode(stamp, bl);
+  encode(pad_size, bl);
```

Comment:
encoding pad_size is unnecessary, because ENCODE_START/ENCODE_END already record length

Reply:
oh, yes, I'll correct that:)

Reply:
ENCODE_START(2, 2, bl);
encode((uint8_t)action, bl);
if (action == NONE) {
encode(stamp, bl);
encode(pad_size, bl);
 uint8_t static const pad = 0xff;
  for (unsigned int i = 0; i<pad_size; i++) {
    encode(pad, bl);
  }
} else {
  encode(ino, bl);
  encode(size, bl);
  encode(stamp, bl);
}